src/Plans.jl

```diff
@@ -92,6 +92,15 @@ Base.IndexStyle(::Plan) = IndexLinear()
 Base.similar(p::Plan) = Plan(p.range, p.varshks, similar(p.exogenous))
 Base.copy(p::Plan) = Plan(p.range, p.varshks, copy(p.exogenous))
 
+function Base.copyto!(dest::Plan,rng::AbstractUnitRange,scr::Plan)
+    @assert dest.varshks === scr.varshks
+    for t in rng
```

Comment:
Instead of running a loop, why not just copy the exogenous data? It'd be more efficient.
```
dest.exogenous[_offset(dest, rng), :] .= src.exogenous[_offset(src, rng), :]
```
Once the `@assert` above has passed, we're guaranteed that the two plans have the same columns in the same order, so a copy like this should work.

Also, because this is going to be called directly by the user, I think it would be better to not use an assertion, but to check with an if-statement and throw an exception (maybe BoundsError?) with a meaningful error message.

Reply:
Even more efficient would be to use copyto! for matrices (avoids the broadcasting machinery of .=)
```
idx2 = axes(dest.exogenous, 2)  # same for both dest and src
copyto!(dest.exogenous, _offset(dest, rng), idx2, src.exogenous, _offset(src, rng), idx2)
```

Reply:
I think ArgumentError would be better than BoundsError if the varshks don't match.

Reply:
Thanks @bbejanov . I implemented your suggestions. I'm not sure what you mean by ArgumentError versus BoundsError. Right now, if the `@assert` fails, we get the following error message.
![plan_assert_err](https://user-images.githubusercontent.com/30166151/199018014-97a1e3bf-1ede-4dcf-9cce-a6442d0225f4.png)

Reply:
It makes complete sense. Thank you for clarifying.